### PR TITLE
introduce proper logging framework, replace panics and printfs

### PR DESCRIPTION
**Pull request by tjungblu**

### CODE REVIEW

File: pkg/obfuscator/ip_test.go [hunk 1]
```diff
@@ -205,15 +205,3 @@ func TestIPObfuscatorConsistent(t *testing.T) {
 		})
 	}
 }
-
-func TestPanicMaximumReplacements(t *testing.T) {
```
Comment (tjungblu) [inline pkg/obfuscator/ip_test.go:209]: that's sadly not testable due to the exit, but I think it's the overall better behaviour than the panic

File: pkg/traversal/traversal.go [hunk 2]
```diff
@@ -84,13 +85,19 @@ func (w *FileWalker) processDir(inputDir input.Directory, outputDirName string)
 			}
 			// If the file should be omitted then stop processing.
 			if omit {
+				if klog.V(2).Enabled() {
+					klog.Infof("omitting '%s'", e.Path())
```
Comment (arjunrn) [inline pkg/traversal/traversal.go:89]: Can't this just be `klog.V(2).Infof()`?

File: pkg/traversal/traversal.go [hunk 3]
```diff
@@ -129,6 +136,10 @@ func (w *FileWalker) processDir(inputDir input.Directory, outputDirName string)
 			if err != nil {
 				return err
 			}
+
+			if klog.V(2).Enabled() {
+				klog.Infof("done processing '%s'", e.Path())
```
Comment (arjunrn) [inline pkg/traversal/traversal.go:141]: And here.

File: pkg/traversal/traversal.go (hunk 2 above)
Comment (tjungblu) [reply pkg/traversal/traversal.go:89]: ah you're right, I was looking at the docs the wrong way. I thought the latter wouldn't skip the locking. Fixing

**Pull request merged**